### PR TITLE
Job Summary API

### PR DESCRIPTION
**Pull request by dadgar**
This PR adds a Job Summary API that returns both the status for the job's task groups and an overall roll up

### CODE REVIEW

File: nomad/job_endpoint.go [hunk 1]
```diff
@@ -516,3 +491,155 @@ func validateJob(job *structs.Job) error {
...
+		Status:     job.Status,
+		TaskGroups: make(map[string]structs.AllocStateCounts, len(job.TaskGroups)),
+	}
+	allocs, err := snap.AllocsByJob(job.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Periodic jobs can't have running instances
+	if job.IsPeriodic() {
```
Comment (diptanu) [inline nomad/job_endpoint.go]: It will be nice to roll up the data from the child jobs of periodic jobs.

File: nomad/job_endpoint.go [hunk 2]
```diff
@@ -516,3 +491,155 @@ func validateJob(job *structs.Job) error {
...
+	}
+
+	// If the job is a system job, pending isn't applicable
+	if job.Type == structs.JobTypeSystem {
+		return status, nil
+	}
+
+	// Pending = Hasn't been scheduled
+	// Determine what is pending by dry-running the scheduler.
+	planner, err := j.dryrunJob(job, snap)
```
Comment (diptanu) [inline nomad/job_endpoint.go]: Can't we get this information from the blocked evals for the job?

File: nomad/job_endpoint.go (hunk 1 above)
Comment (dadgar) [inline nomad/job_endpoint.go]: I am not sure that is desirable? The important thing on periodic jobs is the list of running of children jobs

File: nomad/job_endpoint.go (hunk 2 above)
Comment (dadgar) [inline nomad/job_endpoint.go]: Nope :/ The evaluation doesn't hold any of that information

**Pull request closed without merging**